tests/test_main.py

```diff
@@ -179,7 +179,9 @@ class Model(BaseModel):
     assert Model().c == 0
 
 
-def test_allow_extra():
+def test_allow_extra_deprecated():
+    """Will be removed in a future release"""
```

Comment:
please [capture warnings](https://docs.pytest.org/en/latest/warnings.html) and check check they're right.